swift-sdk/Internal/InAppManager.swift

```diff
@@ -533,18 +488,21 @@ class InAppManager : NSObject, IterableInAppManagerProtocolInternal {
     private let dateProvider: DateProviderProtocol
     private let retryInterval: TimeInterval // in seconds, if a message is already showing how long to wait?
     private var lastDismissedTime: Date? = nil
+    
     private let updateQueue = DispatchQueue(label: "UpdateQueue")
     private let scheduleQueue = DispatchQueue(label: "ScheduleQueue")
-    private let syncQueue = DispatchQueue(label: "SyncQueue")
     private let callbackQueue = DispatchQueue(label: "CallbackQueue")
+    private let syncQueue = DispatchQueue(label: "SyncQueue")
+    
+    private var syncResult: Future<Bool, Error>?
     private var lastSyncTime: Date? = nil
     private let moveToForegroundSyncInterval: Double = 1.0 * 60.0 // don't sync within sixty seconds
 }
 
 extension InAppManager : InAppNotifiable {
-    func onInAppSyncNeeded() {
+    func onInAppSyncNeeded() -> Future<Bool, Error> {
```

Comment:
Since the purpose of the method has changed and it returns a Future to the caller, `onInAppSyncNeeded` isn't a proper name anymore I think.. It should be a verb like `scheduleSync`, etc.